src/OpenTelemetry.Api/BaggageEntry.cs

```diff
@@ -0,0 +1,54 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+namespace OpenTelemetry;
+
+/// <summary>
+/// Baggage entry consisting of a value with optional metadata.
+/// </summary>
+public readonly struct BaggageEntry : IEquatable<BaggageEntry>
```

Comment:
nit: If you define `BaggageEntry` and `BaggageEntryMetadata` as `readonly record struct` the compiler should auto-generate a lot of this boilerplate for you (IEquatable<> implementation, GetHashCode, ToString, operators, etc).